### PR TITLE
add-search-sort-pagination-to-probe-display

### PR DESCRIPTION
**Pull request by lzeee**
### History background
It's an old issue https://github.com/jenkins-infra/plugin-health-scoring/issues/65,  which has been placed for several months and labled as help-wanted. What I have done is just trying to add the necessary packages to the system and make it work. And **90% of the codes comes from 2 historical PR** made by https://github.com/jenkins-infra/plugin-health-scoring/pull/109 @dheerajodha and https://github.com/jenkins-infra/plugin-health-scoring/pull/194 @impact-maker. Thanks for the efforts and the valuable historical code review comments @alecharp .

### Description
This PR give the probe table abilities to perform operations such as sorting the probes by order,  seach probes by any attribute. In order to accomplish this goal, I mainly did 2 things based on the historical code:
1. add 3 npm packages(datatables.net、jquery、expose-loader) and make them work.
2. add 4 *.less files to control the style of 4 added component, which is supplied by datatables.net.

show

### Todo
I rarely wrote front-end code. So I really really need help and code reivew to improve these codes. There must be a lot points to be further discussed to get it done.
1. The style of newly added component is really ugly, which require adjustments.
2. The documentation need improve.
3. This ability is actually general. It can be directly extended to all other tables if necessary.

### Submitter checklist
- [x] The description of this pull request is detailed and explain why this pull request is needed
- [x] The changeset is on a specific branch
  - `feature/` for new feature, or improvements
  - `fix/` for bug fixes
  - `docs/` for any documentation changes
- [ ] If required, the documentation has been updated
- [ ] There is automated tests to cover the code change / addition
    - If there is no test, include an explanation why in the description
- [x] Run `mvn verify` locally and all tests are passing successfully
  - It is OK to create a pull request which has failing tests if it is created as a draft, is to fix a bug and the first commit is the test to prove the existence of the bug.
- [ ] There is no new warnings (checkstyle nor spotbugs) on the code

### CODE REVIEW

**Comment by alecharp** [comment]
I'll review the content, but first, please note that you are using part of the work from @dheerajodha from #65 without attribution of that work. I added the `help-wanted` label on that pull request because someone might want to continue the work and start from it. The branch used for #65 is located in @dheerajodha fork which is publicly available.

**Comment by lzeee** [comment]
@alecharp @alecharp 
Sorry for not starting from the original branch. I have just started to understand the open source community. I mistakenly thought I can just copy the original code as basis, and mention the author's contribution in my description.

**Review by alecharp — CHANGES_REQUESTED** [review]
I really would have preferred that you start your work from https://github.com/dheerajodha/plugin-health-scoring/tree/feature/sort-probes as most of the work in this PR was already done by someone else.

**Comment by lzeee** [comment]
> 

Ok, I will close this PR and make a fork from the original repository and make modification on that branch

I think I understand what you mean by now, because it is very unfair that the original contributor does not leave a trace of contribution.

**Comment by alecharp** [comment]
@lzeee you don't need to create a fork from @dheerajodha repository.
you can create a remote to his repository, fetch the Git data from there and checkout his branch

```
git remote add dheerajodha git@github.com:dheerajodha/plugin-health-scoring.git
git fetch dheerajodha
git switch -c feature/sort-probes dheerajodha/feature/sort-probes
```

Then you can push to your fork and create a pull request from there

```
git push -u origin feature/sort-probes
```

**Comment by lzeee** [comment]
@alecharp 
Thank you very much for the detailed and patient guidance. I'm working on it now.

File: .yarnrc.yml [hunk 1]
```diff
@@ -1 +1,3 @@
+npmRegistryServer: "https://registry.npm.taobao.org/"
+
```
[the remark this answers is not in the document]
Comment (lzeee) [reply .yarnrc.yml:2]: OK, I will remove that

File: src/main/js/index.js [hunk 2]
```diff
@@ -2,3 +2,11 @@ import { defineCustomElements } from "ionicons/dist/loader";
...
   resourcesUrl: '/',
 });
+
+import $ from 'jquery';
+import dt from 'datatables.net';
+$(document).ready( function () {
+  $('#probes-table').DataTable(
+    {"order":[[3, 'asc']]}
+  );
+} );
```
[the remark this answers is not in the document]
Comment (lzeee) [reply src/main/js/index.js:12]: Yes, it should be. I will move this code to the proper place

**Pull request closed without merging**